src/main/java/rs/teslaris/core/repository/document/EventsRelationRepository.java

```diff
@@ -0,0 +1,25 @@
+package rs.teslaris.core.repository.document;
+
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import rs.teslaris.core.model.document.EventsRelation;
+
+@Repository
+public interface EventsRelationRepository extends JpaRepository<EventsRelation, Integer> {
+
+    @Query("select er from EventsRelation er where er.source.id = :eventId")
+    List<EventsRelation> getRelationsForOneTimeEvent(Integer eventId);
+
+    @Query("select er from EventsRelation er where er.target.id = :eventId")
+    List<EventsRelation> getReverseRelationsForOneTimeEvent(Integer eventId);
+
+    @Query("select er from EventsRelation er where er.target.id = :serialEventId")
+    List<EventsRelation> getRelationsForSerialEvent(Integer serialEventId);
```

Comment:
Why do you need these two methods? I think you should keep only the first one or change the second one query to include relation type (belongs to), otherwise there is no sense for me in keeping both methods.

Reply:
You are right, I will combine these 2 into one.